ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/PipeChainTypeHintStage.fs

```diff
@@ -0,0 +1,153 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Stages
+
+open System
+open System.Collections.Generic
+open JetBrains.Application.Settings
+open JetBrains.DocumentModel
+open JetBrains.ProjectModel
+open JetBrains.ReSharper.Daemon.Stages
+open JetBrains.ReSharper.Plugins.FSharp
+open JetBrains.ReSharper.Plugins.FSharp.Daemon.Stages.Tooltips
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Util
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Settings
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Daemon.Cs.Stages
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open FSharp.Compiler.SourceCodeServices
+open FSharp.Compiler.Layout
+open JetBrains.Util.Logging
+
+[<RequireQualifiedAccess>]
+[<Struct>]
+type SameLinePipeHints =
+    | Show
+    | Hide
+
+type PipeOperatorVisitor(fsFile: IFSharpFile, checkResults: FSharpCheckFileResults, sameLinePipeHints: SameLinePipeHints) =
+    inherit TreeNodeVisitor<List<FSharpIdentifierToken * ITreeNode>>()
+
+    let document = fsFile.GetSourceFile().Document
+
+    let showSameLineHints =
+        match sameLinePipeHints with
+        | SameLinePipeHints.Show -> true
+        | SameLinePipeHints.Hide -> false
+
+    let visitBinaryAppExpr binaryAppExpr (consumer: List<FSharpIdentifierToken * ITreeNode>) =
```

Comment:
I'd suggest using `context` name from the base members here, so it doesn't look like daemon/analyzer results consumer.